src/Hosting/Hosting/src/GenericHost/GenericWebHostService.cs

```diff
@@ -73,6 +73,18 @@ public async Task StartAsync(CancellationToken cancellationToken)
                 urls = Options.WebHostOptions.ServerUrls;
             }
 
+            if (string.IsNullOrEmpty(urls))
+            {
+                // HTTP_PORTS and HTTPS_PORTS, these are lower priority than Urls.
+                var httpPorts = Configuration[WebHostDefaults.HttpPortKey];
+                var httpUrls = httpPorts?.Split(';', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries)
```

Comment:
This code should skip non-numeric ports. Needs more validation.

Reply:
How is that different from having invalid values specified in Urls? We don't do any validation on those at this layer. The server does the validation and will throw.

Reply:
I guess so? One difference here is that we're formatting this into URLs so the translation might lead to weird error messages.

Reply:
Here's the error you get for setting either Urls = http://*:BadPort or HTTP_PORTS = BadPort.
A) It's the same.
B) Kestrel should be giving a better error message. I'll file that separately.
```
Unhandled exception. System.Net.Sockets.SocketException (10013): An attempt was made to access a socket in a way forbidden by its access permissions.
   at System.Net.Sockets.Socket.UpdateStatusAfterSocketErrorAndThrowException(SocketError error, Boolean disconnectOnFailure, String callerName)
   at System.Net.Sockets.Socket.DoBind(EndPoint endPointSnapshot, SocketAddress socketAddress)
   at System.Net.Sockets.Socket.Bind(EndPoint localEP)
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransportOptions.CreateDefaultBoundListenSocket(EndPoint endpoint) in D:\github\aspnetcore\src\Servers\Kestrel\Transport.Sockets\src\SocketTransportOptions.cs:line 152
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketConnectionListener.Bind() in D:\github\aspnetcore\src\Servers\Kestrel\Transport.Sockets\src\SocketConnectionListener.cs:line 44
   at Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.SocketTransportFactory.BindAsync(EndPoint endpoint, CancellationToken cancellationToken) in D:\github\aspnetcore\src\Servers\Kestrel\Transport.Sockets\src\SocketTransportFactory.cs:line 40
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.TransportManager.BindAsync(EndPoint endPoint, ConnectionDelegate connectionDelegate, EndpointConfig endpointConfig, CancellationToken cancellationToken) in D:\github\aspnetcore\src\Servers\Kestrel\Core\src\Internal\Infrastructure\TransportManager.cs:line 45
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.<>c__DisplayClass30_0`1.<<StartAsync>g__OnBind|0>d.MoveNext() in D:\github\aspnetcore\src\Servers\Kestrel\Core\src\Internal\KestrelServerImpl.cs:line 207
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context, CancellationToken cancellationToken) in D:\github\aspnetcore\src\Servers\Kestrel\Core\src\Internal\AddressBinder.cs:line 89
   at Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions.BindAsync(AddressBindContext context, CancellationToken cancellationToken) in D:\github\aspnetcore\src\Servers\Kestrel\Core\src\ListenOptions.cs:line 190
   at Microsoft.AspNetCore.Server.Kestrel.Core.AnyIPListenOptions.BindAsync(AddressBindContext context, CancellationToken cancellationToken) in D:\github\aspnetcore\src\Servers\Kestrel\Core\src\AnyIPListenOptions.cs:line 33
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.AddressesStrategy.BindAsync(AddressBindContext context, CancellationToken cancellationToken) in D:\github\aspnetcore\src\Servers\Kestrel\Core\src\Internal\AddressBinder.cs:line 232
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindAsync(IEnumerable`1 listenOptions, AddressBindContext context, CancellationToken cancellationToken) in D:\github\aspnetcore\src\Servers\Kestrel\Core\src\Internal\AddressBinder.cs:line 31
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.BindAsync(CancellationToken cancellationToken) in D:\github\aspnetcore\src\Servers\Kestrel\Core\src\Internal\KestrelServerImpl.cs:line 313
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServerImpl.StartAsync[TContext](IHttpApplication`1 application, CancellationToken cancellationToken) in D:\github\aspnetcore\src\Servers\Kestrel\Core\src\Internal\KestrelServerImpl.cs:line 233
   at Microsoft.AspNetCore.Hosting.GenericWebHostService.StartAsync(CancellationToken cancellationToken) in D:\github\aspnetcore\src\Hosting\Hosting\src\GenericHost\GenericWebHostService.cs:line 149
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at GenericWebHost.Program.Main(String[] args) in D:\github\aspnetcore\src\Hosting\samples\GenericWebHost\Program.cs:line 32
   at GenericWebHost.Program.<Main>(String[] args)

D:\github\aspnetcore\artifacts\bin\GenericWebHost\Debug\net7.0\GenericWebHost.exe (process 12828) exited with code -532462766.
```
The problem is here:
https://github.com/dotnet/aspnetcore/blob/baab05891458c5184a9c14cfcaaa2e223566c9e4/src/Http/Http/src/BindingAddress.cs#L175-L200
It should fail if there's a colon and then an unparsable int, but instead it adds a default port, resulting in scheme: http, host: "*:BadPort", port: 80, and then Kestrel discards the host and binds to 80.

Reply:
Thanks @Tratcher